src/App.tsx

```diff
@@ -1,24 +1,283 @@
+  const loadTodos = async () => {
+    setErrorMessage('');
+
+    try {
+      const dataFromServer = await getTodos();
+
+      setTodos(dataFromServer);
+    } catch (error) {
+      showErrorMessage(ErrorsMessages.Load);
+    }
+  };
+
+  const setFilter = (filter: Filter) => {
+    setCurrentFilter(filter);
+  };
+
+  const filterTodos = (filteringProperty: Filter) => {
+    return (
+      filteringProperty === Filter.all
+        ? todos
+        : todos.filter(todo => {
+          switch (filteringProperty) {
+            case Filter.active:
+              return !todo.completed;
+            case Filter.completed:
+              return todo.completed;
+            default:
+              return null;
+          }
+        })
+    );
+  };
+
+  const filteredTodos: Todo[] = useMemo(
+    () => filterTodos(currentFilter),
+    [todos, currentFilter],
+  );
+
+  const addTodo = async (newTodoTitle: string) => {
+    setErrorMessage('');
+    setIsHeaderDisabled(true);
+
+    try {
+      setTempTodo({
+        id: 0,
+        userId: USER_ID,
+        title: newTodoTitle,
+        completed: false,
+      });
+      const newTodoPost = await createTodo({
+        title: newTodoTitle,
+        userId: USER_ID,
+        completed: false,
+      });
+
+      setTodos((items) => [...items, newTodoPost]);
+    } catch (error) {
+      showErrorMessage(ErrorsMessages.Post);
+    }
+
+    setTempTodo(undefined);
+    setIsHeaderDisabled(false);
+  };
+
+  const filterTodosById = (todoElements: Todo[], id: number) => {
+    return todoElements.filter((item) => item.id !== id);
+  };
+
+  const removeTodo = async (id: number) => {
+    setErrorMessage('');
+    setProcessings((prev) => [...prev, id]);
+
+    try {
+      const response = await deleteTodo(id);
+
+      if (response) {
+        setTodos(filterTodosById(todos, id));
+      }
+    } catch (error) {
+      showErrorMessage(ErrorsMessages.Delete);
+    }
+
+    setProcessings([]);
+  };
+
+  const removeCompletedTodos = async () => {
+    setErrorMessage('');
+
+    const completedTodos = (
+      todos.filter(el => el.completed)
+        .map(item => item.id)
+    );
+
+    setProcessings(completedTodos);
+
+    try {
+      const removed = await Promise.all(
+        completedTodos.map(todo => deleteTodo(todo)),
+      );
+
+      if (removed) {
+        setTodos((prevState) => {
+          return prevState.filter((todo) => !todo.completed);
+        });
+      }
+    } catch (error) {
+      showErrorMessage(ErrorsMessages.Delete);
+    }
+
+    setProcessings([]);
+  };
+
+  const handleChecker = async (id: number, data: Partial<Todo>) => {
+    setErrorMessage('');
+    setProcessings((prev) => [...prev, id]);
+
+    try {
+      const response = await updateTodo(id, data);
+      const todoEdited = todos.find((todo) => todo.id === id);
+      const isTitleChanged = response.title !== todoEdited?.title;
+      const isCompletedChanged = response.completed !== todoEdited?.completed;
```

Comment:
Useless checks. Because if nothing isn't changed so why did you send the request? You should check it first